x/evm/types/params.go

```diff
@@ -176,23 +188,24 @@ func validateChainConfig(i interface{}) error {
 	return cfg.Validate()
 }
 
-func validatePrecompiles(i interface{}) error {
+// ValidatePrecompiles checks if the precompile addresses are valid and unique.
+func ValidatePrecompiles(i interface{}) error {
 	precompiles, ok := i.([]string)
 	if !ok {
 		return fmt.Errorf("invalid precompile slice type: %T", i)
 	}
 
-	seenPrecompiles := make(map[string]bool)
+	seenPrecompiles := make(map[string]struct{})
 	for _, precompile := range precompiles {
-		if seenPrecompiles[precompile] {
+		if _, ok := seenPrecompiles[precompile]; !ok {
 			return fmt.Errorf("duplicate precompile %s", precompile)
 		}
 
 		if err := types.ValidateAddress(precompile); err != nil {
 			return fmt.Errorf("invalid precompile %s", precompile)
 		}
 
-		seenPrecompiles[precompile] = true
+		seenPrecompiles[precompile] = struct{}{}
```

Comment:
nit: the bool approach seemed cleaner

Reply:
this saves memory though, since the empty struct is not actually stored in memory - I saw this in a StackOverflow thread once and used it on the EIP validation, check this:
https://stackoverflow.com/a/9251352